Dev/prplmesh wps backhaul sta onboarding

tomereli opened the pull request:
Ongoing work to enable certification mode wireless onboarding in prplmesh.

Commit:
bwl: add start_wps_pbc API

Add API to start STA wps_pbc in the sta_wlan_hal.
This will be used by the backhaul manager when starting wps onboarding
with wireless backhaul.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>
Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

Commit:
agent: ucc listener remove redundant wps sanity for vaps

When starting wps registration as STA, the VAPs are not yet active.
When starting wps registtration as AP, we don't care if they are active
or not - the command will fail in the backhaul manager.
So 0 remove the redundant check for searching the VAP with the requested
radio mac.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

arnout commented on tools/docker/builder/openwrt/build.sh:
```diff
@@ -117,6 +117,12 @@ main() {
+        axepoint)
+            TARGET_SYSTEM=intel_mips
+            SUBTARGET=xrx500
+            TARGET_PROFILE=DEVICE_AX6000_2000_ETH_11AXUCI
```
(nitpick) since it's called ax6000 here, wouldn't it be better to call the device ax6000?

arnout commented on common/beerocks/bwl/dummy/sta_wlan_hal_dummy.cpp:
```diff
@@ -24,6 +24,8 @@ sta_wlan_hal_dummy::sta_wlan_hal_dummy(const std::string &iface_name, hal_event_
 
 sta_wlan_hal_dummy::~sta_wlan_hal_dummy() { sta_wlan_hal_dummy::detach(); }
 
+bool sta_wlan_hal_dummy::wps_pbc_multiap() { return true; }
```
Please add a log message here so we can test it in test_flows.

arnout commented on common/beerocks/bwl/nl80211/sta_wlan_hal_nl80211.cpp:
```diff
@@ -25,6 +25,8 @@ sta_wlan_hal_nl80211::~sta_wlan_hal_nl80211() { sta_wlan_hal_nl80211::detach();
 
 bool sta_wlan_hal_nl80211::detach() { return true; }
 
+bool sta_wlan_hal_nl80211::wps_pbc_multiap() { return true; }
```
Could you already add the same implementation as dwpal here?

arnout commented on common/beerocks/bwl/nl80211/sta_wlan_hal_nl80211.h:
```diff
@@ -31,6 +31,7 @@ class sta_wlan_hal_nl80211 : public base_wlan_hal_nl80211, public sta_wlan_hal {
     sta_wlan_hal_nl80211(const std::string &iface_name, hal_event_cb_t callback);
     virtual ~sta_wlan_hal_nl80211();
 
+    virtual bool wps_pbc_multiap() override;
```
Why the `_multiap` bit? Yes, we have to add the `multi_ap=1` bit to the command, but in prplmesh we're never going to use anything other than multi_ap, right?

tomereli replied:
Right.

arnout reviewed (commented):
Would be good to already merge the first 5 commits.

Commit:
bcl: socket_thread: handle sd->readBytes() < 0

readBytes return -1 for failure, which is converted to a huge number
since available_bytes type is size_t, which results in failures
afterwards.
Fix it by using ssize_t for available_bytes and handle the case when it
is negative.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
tools: docker: builder openwrt bump feed_wlan_6x

Update feed_wlan_6x dddeadc60130cf90b620b36719b8cfe267555bee:
"swpal: add patch to associate sta interface to br-lan"

With this patch, supplicant will always put STA interfaces in br-lan
(currently hard coded) when they get connected, and remove them when
they get disconnected.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
agent: backhaul manager wps onboarding support

Add wps onboarding initial support by modifying the wireless backhaul
FSM to always wait for WPS. This is good enough for certification, but
not for product, since after reboot the whole process needs to start all
over again.
In addition, the only support added in this patch is for UCC command
which triggers the onboarding.

Summary of changes:
- On WPA_ATTACH, move to WAIT_WPS state without checking for connected
- On Event:Connected, move to MASTER_DISCOVERY

This assumes that the platform is handling the backhaul link, e.g. when
it gets connected it is automatically added to the bridge.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>
Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

rmelotte commented on agent/config/beerocks_agent.conf.in:
```diff
@@ -10,7 +10,7 @@
 temp_path=@TMP_PATH@
 vendor=Intel
-model=prplMesh
+model=AXEPOINT
```
With https://github.com/prplfoundation/prplMesh/pull/1512, this can be removed

arnout reviewed (approved):
Even though the last commit is indeed a hack, it's hard to fix it now because we don't have enough hardware to test it on. Since this is in the critical path for M2, I propose to merge it as is and fix later. I've already created https://jira.prplfoundation.org/browse/PPM-257 for that.

VladyslavTupikin reviewed (approved):
Agree with @arnout.

arnout committed:
gitlab-ci: replace netgear-rax40 bhwifi tests with axepoint

Since wireless backhaul doesn't work on RAX40 and never will, it doesn't
make any sense to have these tests in CI. Instead, add them for
axepoint.

The yml file is created my simply renaming it and replacing
"netgear-rax40" with "axepoint".

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>
